baseclasses/utils/tecplotIO.py

```diff
@@ -0,0 +1,1621 @@
+        self._writeCommonHeader(handle)  # Write the common header information
+
+        # --- Specific to FE Zones ---
+        _writeInteger(handle, self.zone.nNodes)  # Write the number of nodes
+        _writeInteger(handle, self.zone.nElements)  # Write the number of elements
+        _writeInteger(handle, 0)  # iCellDim (future use, set to 0)
+        _writeInteger(handle, 0)  # jCellDim (future use, set to 0)
+        _writeInteger(handle, 0)  # kCellDim (future use, set to 0)
+        _writeInteger(handle, BinaryFlags.FALSE.value)  # No aux data
+
+    def writeFooter(self, handle: TextIO):
+        """Write the zone footer to the file. This includes the connectivity information.
+
+        Parameters
+        ----------
+        handle : TextIO
+            The file handle.
+        """
+        self.zone.connectivity.astype("int32").tofile(handle)
+
+
+class TecplotWriterBinary:
+    def __init__(
+        self,
+        title: str,
+        zones: List[TecplotZone],
+        precision: Literal["SINGLE", "DOUBLE"],
+    ) -> None:
+        """Writer for Tecplot files in binary format.
+
+        Parameters
+        ----------
+        title : str
+            The title of the Tecplot file.
+        zones : List[TecplotZone]
+            A list of Tecplot zones to write.
+        precision : Literal["SINGLE", "DOUBLE"]
+            The floating point precision to write the data.
+        """
+        self.title = title
+        self.zones = zones
+        self.precision = precision
+        self._checkVariables()
+
+    def _checkVariables(self) -> None:
+        """Check that all zones have the same variables."""
+        if not all(set(self.zones[0].variables) == set(zone.variables) for zone in self.zones):
+            raise ValueError("All zones must have the same variables.")
+
+    def _getZoneWriter(self, zone: TecplotZone) -> TecplotZoneWriterBinary:
+        """Get the appropriate zone writer based on the zone type.
+
+        Parameters
+        ----------
+        zone : TecplotZone
+            The Tecplot zone to write.
+
+        Returns
+        -------
+        TecplotZoneWriterBinary
+            The appropriate zone writer object.
+
+        Raises
+        ------
+        ValueError
+            If the zone type is invalid.
+        """
+        if isinstance(zone, TecplotOrderedZone):
+            return TecplotOrderedZoneWriterBinary(self.title, zone, self.precision)
+        elif isinstance(zone, TecplotFEZone):
+            return TecplotFEZoneWriterBinary(self.title, zone, self.precision)
+        else:
+            raise ValueError("Invalid zone type.")
+
+    def write(self, filename: Union[str, Path]) -> None:
+        """Write the Tecplot file to disk.
+
+        Parameters
+        ----------
+        filename : Union[str, Path]
+            The filename as a string or pathlib.Path object.
+        """
+        with open(filename, "wb") as handle:
+            handle.write(b"#!TDV112")  # Magic number
```

Comment:
Dont remember the details, but I think there are others available? This is probably the only relevant, though. However, should this be a defined as a constant upfront?

Reply:
Yes there are other magic numbers, but they correspond to completely different binary layouts. When I was testing the others, I realized tecplot was loading them inconsistently and decided to stick with only this magic number because of stability. I will make it a constant and add a commment that clarifies this is a format specific identifier.